planner/core/rule_decorrelate.go

```diff
@@ -189,6 +189,24 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan, opt *lo
 			}
 			appendRemoveProjTraceStep(apply, proj, opt)
 			return s.optimize(ctx, p, opt)
+		} else if li, ok := innerPlan.(*LogicalLimit); ok {
+			// The presence of 'limit' in 'exists' will make the plan not optimal, so we need to decorrelate the 'limit' of subquery in optimization.
+			// e.g. select count(*) from test t1 where exists (select value from test t2 where t1.id = t2.id limit 1); When using 'limit' in subquery, the plan will not optimal.
+			// If apply is not SemiJoin, the output of it might be expanded even though we are `limit 1`.
+			if apply.JoinType != SemiJoin && apply.JoinType != LeftOuterSemiJoin && apply.JoinType != AntiSemiJoin && apply.JoinType != AntiLeftOuterSemiJoin {
+				goto NoOptimize
+			}
+			// If subquery has some filter condition, we will not optimize limit.
+			if len(apply.LeftConditions) > 0 || len(apply.RightConditions) > 0 || len(apply.OtherConditions) > 0 || len(apply.EqualConditions) > 0 {
+				goto NoOptimize
+			}
+			// If offset is not zero, subquery will become always true or always false. This scenario optimizer has internal optimizations.
```

Comment:
better refine the comment here (eg: limit with non-0 offset will conduct an impact of itself on the final result set from its sub-child, consequently determining the bool value of the Exist subq)

Reply:
well done